code/game/objects/structures/xeno.dm

```diff
@@ -718,3 +718,76 @@ TUNNEL
 			to_chat(M, "<span class='warning'>\The [src] ended unexpectedly, so we return back up.</span>")
 	else
 		to_chat(M, "<span class='warning'>Our crawling was interrupted!</span>")
+
+//Xeno Sensor
+/obj/effect/alien/resin/sensor
+	name = "sensor"
+	desc = "A strange resin structure that seems to be looking around."
+	icon_state = "sensoreye"
+	density = FALSE
+	opacity = FALSE
+	anchored = TRUE
+	max_integrity = 5
+	layer = RESIN_STRUCTURE_LAYER
+
+	var/cooldown = FALSE
+	var/number = 0
+
+/obj/effect/alien/resin/sensor/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/effect/alien/resin/sensor/Destroy()
+	var/area/A = get_area(src)
+	for(var/X in GLOB.alive_xeno_list)
+		to_chat(X, "<span class='xenowarning'>A sensor has been destroyed at [A].</span>")
+	. = ..()
+
+/obj/effect/alien/resin/sensor/process()
+	for(var/mob/living/carbon/H in GLOB.alive_human_list)
+		if(cooldown)
+			return
+		if(H in viewers(src))
+			if(!isxeno(H) && H.stat != DEAD)
+				to_chat(H, "<span class='warning'>You feel something glance at you...</span>")
+				number++
+	var/area/A = get_area(src)
+	for(var/X in GLOB.alive_xeno_list)
+		if(number > 1)
+			var/textnumber = num2text(number)
+			to_chat(X, "<span class='xenowarning'>There are [textnumber] hosts at [A].</span>")
+		else
+			to_chat(X, "<span class='xenowarning'>There is a host at [A].</span>")
```

Comment:
This doest't check for nearby at all

Reply:
it doesnt check for nearby intentionally as its meant to broadcast to the whole hive. Should I make it only in range?

Reply:
don't use process, this won't work like you want.